src/ast/mod.rs

```diff
@@ -122,21 +121,59 @@ impl From<&str> for Ident {
     fn from(value: &str) -> Self {
         Ident {
             value: value.to_string(),
-            quote_style: None,
+            quote_style: QuoteStyle::None,
         }
     }
 }
 
 impl fmt::Display for Ident {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self.quote_style {
-            Some(q) if q == '"' || q == '\'' || q == '`' => {
-                let escaped = value::escape_quoted_string(&self.value, q);
-                write!(f, "{}{}{}", q, escaped, q)
+            QuoteStyle::None => f.write_str(&self.value),
+            QuoteStyle::SingleQuote | QuoteStyle::DoubleQuote | QuoteStyle::BackTick => {
+                let escaped = value::escape_quoted_string(&self.value, self.quote_style.as_char());
```

Comment:
@omer-shtivi I don't get why you need to escape the value to print it.

Can't you just access the self.value? This seems to turn the code more complex to upstream projects

Reply:
@AugustoFKL can't because idents like 
```
"SELECT ```quoted identifier```"
```
will fail
please see the following test:
https://github.com/omer-shtivi/sqlparser-rs/blob/914810d36617581c2380ae9c0f501116a7d810b6/tests/sqlparser_mysql.rs#L476-L477

Reply:
Never saw this syntax, really interesting.

One question. If the column name is `c1` (with the quotes, same as your example), why escape it?

Reply:
Because the column name might be like this:
```
"my_column_""_name"
```
or
```
"my_column_\"_name"
```